mocks/helpers/buildQuerySnapShot.js

```diff
@@ -278,10 +291,15 @@ function _recordsWithOneOfValues(records, key, value) {
   return records.filter(
     record =>
       record &&
-      record[key] &&
-      Array.isArray(record[key]) &&
+      getValueByPath(record, key) &&
+      Array.isArray(getValueByPath(record, key)) &&
       value &&
       Array.isArray(value) &&
-      record[key].some(v => value.includes(v)),
+      getValueByPath(record, key).some(v => value.includes(v)),
   );
 }
+
+function getValueByPath(record, path) {
+  const keys = path.split('.');
+  return keys.reduce((nestedObject, key) => nestedObject?.[key], record);
```

Comment:
The library doesn't currently have optional chaining enabled in the babel presets. Would you mind just updating this line to account for the case that `nestedObject` is undefined?

```suggestion
  return keys.reduce((nestedObject = {}, key) => nestedObject[key], record);
```

Reply:
@sbatson5 fixed, check please